packages/protons/src/index.ts

```diff
@@ -55,6 +60,11 @@ const encoderGenerators: Record<string, (val: string) => string> = {
   uint64: (val) => `w.uint64(${val})`
 }
 
+const encoderGeneratorsJsTypeOverrides: Record<string, (val: string) => string> = {
+  number: (val: string) => `BigInt(${val})`,
```

Comment:
I haven't looked at exactly whats going on here but this doesn't seem ideal.
One goal of this feature is to take advantage of speedups available without using BigInt, but if we're going thru BigInt then that isn't happening.

Somehow I thought this feature would require creating some additional types eg `w.number64` where we could have optimized encoder/decoders.

Reply:
Ok, I've ported the original `LongBits.fromNumber` method from `protobuf.js` and split the codepaths out for bigints, numbers and strings so we have single call sites for future optimisations.